cmd/eksctl-anywhere/cmd/validate.go

```diff
@@ -0,0 +1,207 @@
+	}
+
+	runner.Register(cmdvalidations.PackageKubeConfigPath(clusterConfig.Name)...)
+
+	if clusterConfig.Spec.DatacenterRef.Kind == v1alpha1.TinkerbellDatacenterKind {
+		flag := cmd.Flags().Lookup(TinkerbellHardwareCSVFlagName)
+
+		// If no flag was returned there is a developer error as the flag has been removed
+		// from the program rendering it invalid.
+		if flag == nil {
+			runner.ReportResults()
+			panic("'hardwarefile' flag not configured")
+		}
+
+		if !viper.IsSet(TinkerbellHardwareCSVFlagName) || viper.GetString(TinkerbellHardwareCSVFlagName) == "" {
+			return runner.ExitError(fmt.Errorf("required flag \"%v\" not set", TinkerbellHardwareCSVFlagName))
+		}
+
+		if !validations.FileExists(cc.hardwareCSVPath) {
+			return runner.ExitError(fmt.Errorf("hardware config file %s does not exist", cc.hardwareCSVPath))
+		}
+	}
+
+	err = runner.StoreValidationResults()
+	if err != nil {
+		return runner.ExitError(err)
+	}
+
+	clusterSpec, err := newClusterSpec(valOpt.clusterOptions)
+	if err != nil {
+		return runner.ExitError(err)
+	}
+
+	cliConfig := buildCliConfig(clusterSpec)
+	dirs, err := valOpt.directoriesToMount(clusterSpec, cliConfig)
+	if err != nil {
+		return runner.ExitError(err)
+	}
+
+	deps, err := dependencies.ForSpec(ctx, clusterSpec).WithExecutableMountDirs(dirs...).
+		WithKubectl().
+		WithProvider(valOpt.fileName, clusterSpec.Cluster, valOpt.skipIpCheck, valOpt.hardwareCSVPath, true, valOpt.tinkerbellBootstrapIP).
+		WithGitOpsFlux(clusterSpec.Cluster, clusterSpec.FluxConfig, cliConfig).
+		Build(ctx)
+	if err != nil {
+		return runner.ExitError(err)
+	}
+	defer close(ctx, deps)
+
+	runner.Register(cmdvalidations.PackageSupportedProvider(deps.Provider)...)
+	err = runner.StoreValidationResults()
+	if err != nil {
+		return runner.ExitError(err)
+	}
+
+	var cluster *types.Cluster
+	if clusterSpec.ManagementCluster == nil {
+		cluster = &types.Cluster{
+			Name:           clusterSpec.Cluster.Name,
+			KubeconfigFile: kubeconfig.FromClusterName(clusterSpec.Cluster.Name),
+		}
+	} else {
+		cluster = &types.Cluster{
+			Name:           clusterSpec.ManagementCluster.Name,
+			KubeconfigFile: clusterSpec.ManagementCluster.KubeconfigFile,
+		}
+	}
+
+	validationOpts := &validations.Opts{
+		Kubectl: deps.Kubectl,
+		Spec:    clusterSpec,
+		WorkloadCluster: &types.Cluster{
+			Name:           clusterSpec.Cluster.Name,
+			KubeconfigFile: kubeconfig.FromClusterName(clusterSpec.Cluster.Name),
+		},
+		ManagementCluster: cluster,
+		Provider:          deps.Provider,
+		CliConfig:         cliConfig,
+	}
+
+	createValidations := createvalidations.New(validationOpts)
+
+	runner.Register(cmdvalidations.PackageCreatePreflight(ctx, createValidations)...)
+	runner.Register(cmdvalidations.PackageProviderValidations(ctx, clusterSpec, deps.Provider)...)
+	runner.Register(deps.GitOpsFlux.Validations(ctx, clusterSpec)...)
+
+	err = runner.StoreValidationResults()
+	runner.ReportResults()
+
+	cleanup(deps, &err)
+	deps.Writer.CleanUp()
```

Comment:
This would unintentionally remove any existing folder and contents sharing the cluster name right?